compiler/z/codegen/ConstantDataSnippet.hpp

```diff
@@ -47,7 +47,7 @@ class S390ConstantDataSnippet : public TR::Snippet
    protected:
    union
       {
-      uint8_t                       _value[16];
+      uint8_t                       _value[64];
```

Comment:
@fjeremic  I have changed the size of _value array with `1 << TR_DEFAULT_DATA_SNIPPET_EXPONENT` to link them. With regard to your question, We can only allocate Data snippet in power of 2 and in method, writable snippets are emitted in order of decreasing size. So `constantDataSnippetExponent`to get the maximum size of snippet and we start from that point.